file/writable_file_writer.cc

```diff
@@ -220,6 +220,9 @@ IOStatus WritableFileWriter::Close() {
       if (ShouldNotifyListeners()) {
         auto finish_ts = FileOperationInfo::FinishNow();
         NotifyOnFileTruncateFinish(start_ts, finish_ts, s);
+        if (!interim.ok()) {
```

Comment:
I am not sure this is related to this specific change but this code looks generally broken to me.  The NotifyOnFiileTruncateFinish passes in the status "s", but that status is not the status of Flush() at line 205.  The same is true later for NotifyOnFileSyncFinish and NotifyOnFileCloseFinish.

Reply:
Yes. It doesn't seems consistent. Here we are passing the status of Flush() but at other places we are passing the status returned by FileSystem Operation.